katz/src/main/kotlin/katz/NonEmptyList.kt

```diff
@@ -66,6 +66,6 @@ class NonEmptyList<out A> private constructor(
     }
 
     companion object Factory {
-        fun <A> of(head: A, vararg t: A): NonEmptyList<A> = NonEmptyList(head, t.asList())
+        fun <A: Any> of(head: A, vararg t: A): NonEmptyList<A> = NonEmptyList(head, t.asList())
     }
 }
```

Comment:
The idea is to name `null` as an unwelcome citizen. This system accepts nullable types but in the case, the type will be null the system crash in the moment that the null appears. Fail fast is better IMHO. For the interoperability, I think the way is use method like `fromNullable` on model with types like Option.